### PR TITLE
Bump upstream to 5.1.1.

### PR DESCRIPTION
**Pull request by Salamandar**
Everything's in the title.

EDIT : tested on my server, works fine.

### CODE REVIEW

**Comment by Salamandar** [comment]
!testme

**Review by ericgaspar — APPROVED** [review]

**Comment by ericgaspar** [comment]
Do you have access to `voir la BD explicative` pages ? I get `The image could not be loaded`

**Comment by Salamandar** [comment]
Hah didnt try it, I thought it was an external link.

**Comment by Salamandar** [comment]
Indeed, it's missing the subdir prefix. Hmmm.

EDIT : It's 200% a bug upstream. `/static/showcase/1.jpg` is hard-coded whereas other static files are referenced by `url_for("static", filename="")`

**Comment by Salamandar** [comment]
@ericgaspar I added a patch, it should be OK :)

**Comment by Salamandar** [comment]
!testme

File: scripts/install [hunk 1]
```diff
@@ -114,6 +114,9 @@ sub_path_only="$(if [[ "$path_url" == "/" ]]; then echo '# ' ; else echo ''; fi)
 ynh_add_config --template ../conf/ihatemoney.cfg --destination "$final_path/ihatemoney.cfg"
 chmod 640 "$final_path/ihatemoney.cfg"
 
+# Upstream issue : https://github.com/spiral-project/ihatemoney/issues/973
+src_dir=$("$final_path/venv/bin/python3" -c "import ihatemoney ; print(ihatemoney.__file__)")
+patch -p1 -d "$(dirname "$src_dir")" < "../conf/ihatemoney.patch"
```
Comment (ericgaspar) [inline scripts/install]:
```suggestion
patch -p1 -d "$(dirname "$src_dir")" < "../conf/ihatemoney_showcase.patch"
```

File: scripts/upgrade [hunk 2]
```diff
@@ -166,6 +166,9 @@ sub_path_only="$(if [[ "$path_url" == "/" ]]; then echo '# ' ; else echo ''; fi)
 ynh_add_config --template ../conf/ihatemoney.cfg --destination "$final_path/ihatemoney.cfg"
 chmod 600 "$final_path/ihatemoney.cfg"
 
+# Upstream issue : https://github.com/spiral-project/ihatemoney/issues/973
+src_dir=$("$final_path/venv/bin/python3" -c "import ihatemoney ; print(ihatemoney.__file__)")
+patch -p1 -d "$(dirname "$src_dir")" < "../conf/ihatemoney.patch"
```
Comment (ericgaspar) [inline scripts/upgrade]:
```suggestion
patch -p1 -d "$(dirname "$src_dir")" < "../conf/ihatemoney_showcase.patch"
```

**Review by ericgaspar — CHANGES_REQUESTED** [review]

File: scripts/upgrade (hunk 2 above)
Comment (ericgaspar) [reply scripts/upgrade]: Also, if you put your patch in` sources -> patches `folder, the patch should be applied automatically after source install.

https://github.com/YunoHost/yunohost/blob/71a08c09ea5099d7951d8d7badf0ed3bc0a02cf2/data/helpers.d/utils#L214-L226

Comment (Salamandar) [reply scripts/upgrade]: I won't put it in the patches folder, because I don't use ynh_setup_source. ihatemoney is installed by pip in the venv :)

Comment (Salamandar) [reply scripts/upgrade]: Good catch for the filename though, I'm fixing it right now.

**Comment by Salamandar** [comment]
Haaaaa there's no easy way to "not fail" if the patch is already applied (that's the case during an upgrade because pip won't replace edited files).
Do you have any trick ?

**Comment by yalh76** [comment]
already at 5.2.0~ynh1

**Pull request closed without merging**